tests/rptest/scale_tests/audit_log_test.py

```diff
@@ -0,0 +1,412 @@
+from ducktape.errors import TimeoutError
+
+# How much memory to assign to redpanda per partition. Redpanda will be started
+# with MIB_PER_PARTITION * PARTITIONS_PER_SHARD * CORE_COUNT memory
+DEFAULT_MIB_PER_PARTITION = 4
+
+# How many partitions we will create per shard: this is the primary scaling
+# factor that controls how many partitions a given cluster will get.
+PARTITIONS_PER_SHARD = 1000
+
+
+class AuditLogTestSecurityConfig(SecurityConfig):
+    """
+    Hardcoded security parameters for the audit scale tests. SASL will be
+    the authentication method of choice
+    """
+    sasl_credentials = SaslCredentials('rob', '123rob456', 'SCRAM-SHA-256')
+
+    def __init__(self):
+        super(AuditLogTestSecurityConfig, self).__init__()
+        self.enable_sasl = True
+        self.kafka_enable_authorization = True
+        self.endpoint_authn_method = 'sasl'
+        self.tls_provider = None
+
+
+class AuditBenchmarkResults():
+    def __init__(self, time_elapsed: float, final_totals: tuple[int, int],
+                 latencies: tuple[float, float, float], message_size: int):
+        self.time_elapsed = time_elapsed
+        self.messages_produced = final_totals[0]
+        self.messages_consumed = final_totals[1]
+        self.message_size = message_size
+        self.p50 = latencies[0]
+        self.p90 = latencies[1]
+        self.p99 = latencies[2]
+
+    def _mbps(self, b: float) -> float:
+        return (b / self.time_elapsed) / (1024 * 1024.0)
+
+    @property
+    def produce_mbps(self) -> float:
+        bytes_produced = self.messages_produced * self.message_size
+        return self._mbps(bytes_produced)
+
+    @property
+    def consume_mbps(self) -> float:
+        bytes_consumed = self.messages_consumed * self.message_size
+        return self._mbps(bytes_consumed)
+
+    @property
+    def total_mbps(self) -> float:
+        total_bytes_moved = (self.messages_produced +
+                             self.messages_consumed) * self.message_size
+        return self._mbps(total_bytes_moved)
+
+    def __repr__(self):
+        return str({
+            'time_elapsed': self.time_elapsed,
+            'total_produced': self.messages_produced,
+            'total_consumed': self.messages_consumed,
+            'p50': self.p50,
+            'p90': self.p90,
+            'p99': self.p99
+        })
+
+
+class AuditLogTest(RedpandaTest):
+    """
+    Test to ensure that enabling auditing does not create a discernable decrease
+    in performance.
+    """
+    topics = ()
+
+    def __init__(self, ctx, *args, **kwargs):
+        # Ensure redpanda is setup with sasl_enabled and endpoints are configured
+        # to work with SASL, otherwise audit client will not be able to connect
+        self.security = AuditLogTestSecurityConfig()
+
+        # We will send huge numbers of messages, so tune down the log verbosity
+        # as we will be comparing performance to a known good baseline
+        kwargs['log_level'] = 'info'
```

Comment:
question: Did you want to set auditing facility to `trace`?

Reply:
it would produce a ton of log messages maybe too many? Not really necessary IMO since we are now periodically printing the metrics stuff , that will help us understand whats happening

Reply:
will change to debug

Reply:
Ah I see, we're just going to collect on when events cannot be inserted, is that it?